packages/core/useWebSocket/index.ts

```diff
@@ -315,7 +315,10 @@ export function useWebSocket<Data = any>(
   if (immediate)
     open()
 
-  watch(urlRef, open)
+  watch(urlRef, () => {
```

Comment:
since the options can't be a ref, could we just avoid setting a watcher in the first place?

i.e.

```suggestion
if (immediate) {
  watch(urlRef, open)
}
```

afaict `immediate` can't change

Reply:
Actually, what you suggested matches exactly how it was in the [initial version](https://github.com/vueuse/vueuse/pull/2367). I'm not sure what happened in between that led to the current situation, but there appears to be some discrepancy between the current functionality and the documentation. 

If this proposal is approved, I plan for `immediate` to be ref/computed as the next step.

Reply:
it looks like it was accidentally lost in 93b96ae0a

if your plan is to make `immediate` a ref anyway though, this code makes sense 👍

Reply:
Ah, indeed, `immediate` is in the concept of the watch's immediate behavior. Reactivity to the URL changes is a different concept. I changed this PR to add another option to control that.